specification/edgemarketplace/Microsoft.EdgeMarketPlace.Management/offer.tsp

```diff
@@ -0,0 +1,66 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+
+namespace Microsoft.EdgeMarketplace;
+@doc("An offer.")
+model Offer is ExtensionResource<OfferProperties> {
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-name-pattern" "Legacy"
+  @doc("Id of the offer")
+  @path
+  @key("offerId")
+  @segment("offers")
+  name: string;
+}
+
+#suppress "@azure-tools/typespec-azure-resource-manager/improper-subscription-list-operation" "backcompat, ARM reviewed"
+@armResourceOperations
+interface Offers {
+  @doc("Get a Offer")
+  get is ArmResourceRead<Offer>;
+  @doc("List Offer resources by parent")
+  list is ArmResourceListByParent<
```

Comment:
since you are defining an extension resource, we should use `Extension.Read<Offer>` here for better semantic.
And if you change the list to `Extension.ListByTarget`, we could also remove the suppress above because there should not be a warning.